tests/test_migrations.py

```diff
@@ -0,0 +1,129 @@
+from cgi import test
+from hashlib import new
+from alembic.config import main as alembic_main
+import pytest
+import json
+
+from mds.config import DB_DSN, DEFAULT_AUTHZ_STR
+from mds.models import db
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "old_metadata, new_metadata, authz_data",
+    [
+        # standard resource_paths should get migrated into `authz` column with standard version number (0)
+        (
+            {"foo": "bar", "_resource_paths": ["/programs/DEV"]},
+            '{"foo": "bar"}',
+            '{"version": 0, "_resource_paths": ["/programs/DEV"]}',
+        ),
+        # default authz resource_paths do not get migrated into `data` column.
+        ({"foo": "bar"}, '{"foo": "bar"}', DEFAULT_AUTHZ_STR),
+        # handle null `data` columns (with populated `authz` columns)
+        (
+            {"_resource_paths": ["/programs/DEV"]},
+            None,
+            '{"version": 0, "_resource_paths": ["/programs/DEV"]}',
+        ),
+    ],
+)
+async def test_4d93784a25e5_downgrade(
+    old_metadata: dict, new_metadata: str, authz_data: str
+):
+    """
+    We can't create metadata by using the `client` fixture because of two reasons:
+    1) the `authz` non-nullable column will not exist in the metadata table after the alembic downgrade
```

Comment:
i don't understand what the problem is here

Reply:
Something like this might be a better description:

Calling the API when the db is in the downgraded state will give UndefinedColumnErrors (metadata.authz) because the db and the data model are out of sync. 

This applies to GET as well as POST requests.

Reply:
oh that makes sense